src/types.js

```diff
@@ -129,7 +134,7 @@ export type Manifest = {
   files?: Array<string>,
   main?: string,
 
-  workspaces?: Array<string>,
+  workspaces?: Array<string> | WorkspacesConfig,
```

Comment:
any reason to keep the old structure?
Maybe use WorkspacesConfig for hoist cases too?

Reply:
this is merely for backward compatibility so people who didn't need the new feature doesn't need to be forced to update their package.json... Internally the code is all converted to use WorkspaceConfig.

Reply:
Ah, ok, let's keep it backwards compat, I missed that this is read from package.json